更换版本接口&界面

Bosheng0422 opened the pull request:
改动SeverUtil里getVersionInfo到新接口
改动update页里单一下载地址为多重并增加弹窗显示

Bosheng0422 committed:
重接版本信息接口，改进update页

Bosheng0422 committed:
更改版本更新提醒弹窗显示历史版本信息

Bosheng0422 committed:
✨ 把版本更新信息弹窗显示历史信息

Yun-Shan commented on src/common/util/ServerUtil.js:
```diff
@@ -441,8 +441,10 @@ export default class ServerUtil {
     const arg = targetVersion ? `?version=${targetVersion}` : '';
-    data = await ServerUtil.requestApi('GET', 'canteen/operate/version/plugin' + arg).catch((error) => {
+
+    data = await ServerUtil.requestCdnServerApi('/cdn/operate/version/fetch' + '?platform=plugin').catch((error) => {
```
字符串合并， 494行同理

Yun-Shan commented on src/components/UpdateInfoNotice.vue:
```diff
@@ -18,30 +32,104 @@ export default {
+          this.lastpage = true;
+        }
+      });
+    },
```
loading没用到 应该这边要写上的，界面的话也可以考虑加loading特效 不加也行

Yun-Shan commented on src/update/App.vue:
```diff
@@ -32,19 +32,42 @@
+
+            <div v-if="selectedItem.spare_urls">
+              <div v-for="(spare, spareIndex) in selectedItem.spare_urls" :key="spareIndex" class="button-wrapper">
+                <el-button type="primary" @click="openUrl(selectedItem.spare_urls[spareIndex].url)"
```
`selectedItem.spare_urls[spareIndex].url`感觉可以改成`spare.url`

Yun-Shan commented on src/update/App.vue:
```diff
@@ -64,11 +87,16 @@ export default {
   data() {
     return {
+      dialogVisible: false, // 控制弹窗显示
+      selectedUrl: '', // 存储选中的 primary_url
```
这个变量好像只有赋值，没有使用

Yun-Shan commented on src/update/App.vue:
```diff
@@ -64,11 +87,16 @@ export default {
       currentVersion: CURRENT_VERSION,
       updateInfo: {},
       isLatestVersion: false,
+      isDialogVisible: false, // 控制弹窗的显示与隐藏
```
`dialogVisible`和`isDialogVisible`重复了，如果是同一个弹窗的话就合并成一个变量，不同弹窗的话就改个更清晰的命名

Yun-Shan commented on src/update/App.vue:
```diff
@@ -88,8 +116,31 @@ export default {
+    handleConfirm() {
+      // 弹窗确认操作逻辑
+      this.isDialogVisible = false;
+      console.log('确认按钮被点击');
```
一共有5个console.log/error调用，如果不需要的话可以移除，如果需要的话建议改为调用`DebugUtil.debugLog`/`DebugUtil.debugLogError`输出(第一个参数是调试等级，如果是需要无条件输出的就用0，如果只在调试模式输出的话用1就可以)

(虽然项目里确实还有很多遗留的console.log没改，不过新代码的话尽量就统一一下调用入口吧

phidiaLam commented on src/common/util/ServerUtil.js:
```diff
@@ -441,8 +441,10 @@ export default class ServerUtil {
     let data;
+
     const arg = targetVersion ? `?version=${targetVersion}` : '';
-    data = await ServerUtil.requestApi('GET', 'canteen/operate/version/plugin' + arg).catch((error) => {
```
targetVersion没用到，接口是支持查特定版本的，还是要加上

phidiaLam replied:
参数带version就是查特定版本，不带查最新

Bosheng0422 replied:
好 把platform和version参数都抽出来了

phidiaLam commented on src/common/util/ServerUtil.js:
```diff
@@ -480,4 +482,52 @@ export default class ServerUtil {
     return data;
   }
+
+  /**
+   * @param checkVersionUpdate {boolean} 是否检测版本更新并推送
+   * @param targetVersion {string} 要获取的目标版本信息，不提供时获取最新版
+   * @param pageId {string} 要获取的页面id，首页没有
+   */
+  static async getVersionHistory(checkVersionUpdate = true, targetVersion = undefined, pageId = undefined) {
```
改注释，还有参数里面的targetVersion应该是用不到吧

phidiaLam commented on src/common/util/ServerUtil.js:
```diff
@@ -480,4 +482,52 @@ export default class ServerUtil {
+   */
+  static async getVersionHistory(checkVersionUpdate = true, targetVersion = undefined, pageId = undefined) {
+    await new Promise((resolve) => Settings.doAfterInit(() => resolve()));
+    pageId = pageId === undefined ? '' : '&first_id=' + pageId;
```
最好别这样子复用pageId，会语意不明，重新搞个const变量接

phidiaLam commented on src/common/util/ServerUtil.js:
```diff
@@ -480,4 +482,52 @@ export default class ServerUtil {
+          checkVersionUpdate = false;
+          return response.text();
+        }
+        console.log(error);
```
这些应该是都不要了吧，版本列表接口不作为检查是否有新版本

phidiaLam commented on src/common/util/ServerUtil.js:
```diff
@@ -480,4 +482,52 @@ export default class ServerUtil {
+          'update'
+        );
+      }
+    }
```
包括这边，版本列表接口不作为检查是否有新版本吧

phidiaLam replied:
整个方法自己检查一下

phidiaLam commented on src/components/UpdateInfoNotice.vue:
```diff
@@ -1,11 +1,25 @@
+    <div ref="observeContainer" class="update-info-area">
+      {{ updateInfo.list[0].description }}
+
+      <h3>{{ '历史版本翻新:' }}</h3>
+      <div v-for="(info, index) in updateInfo.list.slice(1)" :key="index">
+        <h4>{{ 'V' + info.version }}</h4>
```
一般用h3 h4这种标签吗？ @Yun-Shan 
我记得是不什么用的，换成css？

Yun-Shan replied:
> 一般用h3 h4这种标签吗？ @Yun-Shan 我记得是不什么用的，换成css？

确实见得少，不用的主要原因是避免不同浏览器实现不同导致的样式区别(但应该没有很大区别)。考虑到这里应该是一块纯文字展示，略有样式区别影响不大(甚至主流的几个可能没区别)，但也可以改用css确保样式统一

Bosheng0422 replied:
换了 样式也改了一下：P

phidiaLam commented on src/components/UpdateInfoNotice.vue:
```diff
@@ -1,11 +1,25 @@
+      <div
+        id="bottom-checker"
+        ref="bottomChecker"
+        style="margin-top: -50px; width: 100%; height: 50px; pointer-events: none"
```
有id了，或者搞个class写到css？

phidiaLam commented on src/components/UpdateInfoNotice.vue:
```diff
@@ -18,30 +32,104 @@ export default {
       let versionUpdate = await PlatformHelper.Storage.getLocalStorage('version-update');
       if (!versionUpdate || CURRENT_VERSION !== versionUpdate) {
-        let data = await ServerUtil.getVersionInfo(false, CURRENT_VERSION);
+        let data = await ServerUtil.getVersionHistory(false, CURRENT_VERSION);
```
这个方法应该不用传参吧

phidiaLam replied:
然后这个方法应该是可以改成调用loadData的

phidiaLam commented on src/components/UpdateInfoNotice.vue:
```diff
@@ -18,30 +32,104 @@ export default {
+        } else {
+          console.error('data.list 不是数组', res.list);
+        }
+        //检测是否最后一页
+        if (res.next_id) {
+          this.nextPageId = res.next_id;
+        } else {
+          this.lastpage = true;
```
这边可以直接赋值给nextPageId，然后下次进来的时候判断nextPageId是null还是有值来判断继续，就不需要lastpage这个变量了。顺带一提，要的话P大写。统一下小驼峰

Bosheng0422 committed:
Review fix ver1.

Bosheng0422 committed:
🚧 review-fix v2
update 页
修改变量spare.url
删除冗余变量selectedUrl
删除冗余变量isDialogVisible以及冗余函数showDetails
删掉了下载弹窗的关闭footer 考虑到与右上x关闭冗余

Bosheng0422 committed:
🚧 V3 update修改
#108 review-fix-v3
修改 getVersionHistory，去掉不必要状态码check和弹窗

phidiaLam committed:
🐛 修复版本更新初始化时候获取不到dom节点

phidiaLam committed:
🐛 修复版本列表刷新生效，修复css样式

Bosheng0422 committed:
🎨 修改console格式为Debugutil

phidiaLam commented on src/common/util/ServerUtil.js:
```diff
@@ -480,4 +483,17 @@ export default class ServerUtil {
+   * @param platform {string} 要获取的端平台 `desktop`、`pocket`、`plugin` 无指定则插件端
+   * @param pageId {string} 要获取的页面id，首页没有
+   */
+  static async getVersionHistory(platform = 'plugin', pageId = undefined) {
```
pageId 建议放前面，这样子platform就不用给了，不然你给pageId时候还得给platform，用不了默认

Bosheng0422 committed:
🐛 修复版本信息title显示失败

Bosheng0422 committed:
🎨 改进参数顺序

Bosheng0422 committed:
🐛 修复参数调用顺序问题，优化参数结构

Bosheng0422 committed:
🎨 改进参数顺序

phidiaLam commented on src/common/util/ServerUtil.js:
```diff
@@ -488,7 +488,7 @@ export default class ServerUtil {
    * @param pageId {string} 要获取的页面id，首页没有
    */
-  static async getVersionHistory(platform = 'plugin', pageId) {
+  static async getVersionHistory(pageId, platform = 'plugin') {
```
这个的调用没有改

Bosheng0422 committed:
🐛 改进调用参数
